addons/mass_mailing/static/tests/mass_mailing_favourite_filter_tests.js

```diff
@@ -407,5 +407,72 @@ QUnit.module('favorite filter widget', (hooks) => {
         assert.isNotVisible(fixture.querySelector('.o_mass_mailing_remove_filter'),
             "should not have option to remove filter because mailing domain is changed");
     });
+
+    QUnit.test('mailing_model_name can be empty without throwing an error', async (assert) => {
+        assert.expect(1);
+
+        serverData.models.partner = {
+            fields: {
+                name: { string: 'Name', type: 'char', searchable: true },
+            },
+            records: [
+                { id: 1, name: 'Azure Interior' },
+                { id: 2, name: 'Deco Addict' },
+                { id: 3, name: 'Marc Demo' },
+            ]
+        };
+
+        serverData.models['mailing.filter'].records = [{
+            id: 1,
+            name: 'Azure Partner Only',
+            mailing_domain: "[['name','=', 'Azure Interior']]",
+            mailing_model_id: 2,
+        }];
+
+        serverData.models['mailing.mailing'].records.push({
+            id: 3,
+            display_name: 'Partner Event promotion',
+            subject: 'Early bird discount for Partners!',
+            mailing_model_id: 2,
+            mailing_model_name: '',
+            mailing_filter_count: 1,
```

Comment:
Set data as if we were using the filter we created
```suggestion
            mailing_filter_count: 1,
            mailing_filter_domain: "[['name','!=', 'Azure Interior']]",
            mailing_filter_id: 1,
```